### PR TITLE
[ie/adn] domain change from .fr/.de to .com

### LINKED ISSUE #10442: ADN now is .com
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

France

### Provide a description that is worded well enough to be understood

ADN now redirect to .com adress, so add this in the extractor.

The .fr work well.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-P', 'E:\\YTdlp\\ADN', '--username', 'PRIVATE', '--password', 'PRIVATE', '-f', 'bv+ba/b', '--sub-langs', 'fr-FR,fr', '--embed-subs', '--embed-chapters', '--remux-video', 'mkv', '-Uv', '-a', 'C:\\Users\\micka\\Downloads\\Google Drive\\Privé\\Plex\\dl_adn.txt']
[debug] Batch file urls: ['https://animationdigitalnetwork.com/video/1198-suicide-squad-isekai/25975-episode-5']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version stable@2024.07.09 from yt-dlp/yt-dlp [7ead7332a] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 1.1.1k  25 Mar 2021) 
[debug] exe versions: ffmpeg n7.0-12-gcbd98447bc-20240420 (setts), ffprobe n7.0-12-gcbd98447bc-20240420 
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1834 extractors 
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.07.09 from yt-dlp/yt-dlp 
yt-dlp is up to date (stable@2024.07.09 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://animationdigitalnetwork.com/video/1198-suicide-squad-isekai/25975-episode-5 
[generic] 25975-episode-5: Downloading webpage
WARNING: [generic] Falling back on generic information extractor 
[generic] 25975-episode-5: Extracting information 
[debug] Looking for embeds
ERROR: Unsupported URL: https://animationdigitalnetwork.com/video/1198-suicide-squad-isekai/25975-episode-5 
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1622, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1757, in __extract_info
  File "yt_dlp\extractor\common.py", line 740, in extract
  File "yt_dlp\extractor\generic.py", line 2521, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://animationdigitalnetwork.com/video/1198-suicide-squad-isekai/25975-episode-5
```

### PR DESCRIPTION
**Pull request by infanf**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

ADN (Animation Digital Network) changed their domain to animationdigitalnetwork.com and removed local domains .fr and .de

Fixes #10442

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by infanf** [comment]
The new domain will go live tomorrow, will test if everything works and if needed fix accordingly then

**Comment by Deluvi** [comment]
Hi, your PR does not quite solve the issue yet, as the ADN's URLs does not include a `/fr/` (at least in France)

I opened a PR on your fork with the changes I made to download some content on my side. Feel free to get the changes you need and close it: https://github.com/infanf/yt-dlp/pull/1

**Comment by infanf** [comment]
> I opened a PR on your fork with the changes I made to download some content on my side. Feel free to get the changes you need and close it: [infanf#1](https://github.com/infanf/yt-dlp/pull/1)

Thank for the fix, but this will break the German site. Yesterday .com redirected to .com/de when in Austria/Germany and to .com/fr otherwise, which is why I made these changes. Looks like the German content is under .com/de and the French under .com only. Adjusted accordingly.

**Comment by Deluvi** [comment]
Works well, cheers

File: yt_dlp/extractor/adn.py [hunk 1]
```diff
@@ -48,9 +48,9 @@ class ADNBaseIE(InfoExtractor):
 
 
 class ADNIE(ADNBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.(?P<lang>fr|de)/video/[^/?#]+/(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.com/((?P<lang>de)/)?video/[^/?#]+/(?P<id>\d+)'
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:51]: non-capturing group for the outer group
```suggestion
    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.com/(?:(?P<lang>de)/)?video/[^/?#]+/(?P<id>\d+)'
```

File: yt_dlp/extractor/adn.py [hunk 2]
```diff
@@ -298,9 +300,9 @@ def _real_extract(self, url):
 
 
 class ADNSeasonIE(ADNBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.(?P<lang>fr|de)/video/(?P<id>[^/?#]+)/?(?:$|[#?])'
+    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.com/((?P<lang>de)/)?video/(?P<id>[^/?#]+)/?(?:$|[#?])'
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:303]: same as above
```suggestion
    _VALID_URL = r'https?://(?:www\.)?(?:animation|anime)digitalnetwork\.com/(?:(?P<lang>de)/)?video/(?P<id>[^/?#]+)/?(?:$|[#?])'
```

File: yt_dlp/extractor/adn.py [hunk 3]
```diff
@@ -327,8 +331,9 @@ def _real_extract(self, url):
 
         def entries():
             for episode_id in traverse_obj(episodes, ('videos', ..., 'id', {str_or_none})):
+                baseurl = 'https://animationdigitalnetwork.com' + ('/' + lang if lang != 'fr' else '')
                 yield self.url_result(
-                    f'https://animationdigitalnetwork.{lang}/video/{video_show_slug}/{episode_id}',
+                    f'{baseurl}/video/{video_show_slug}/{episode_id}',
                     ADNIE, episode_id)
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:337]:
```py
                yield self.url_result(join_nonempty(
                    'https://animationdigitalnetwork.com', lang, 'video',
                    video_show_slug, episode_id, delim='/'), ADNIE, episode_id)
```

File: yt_dlp/extractor/adn.py [hunk 4]
```diff
@@ -177,6 +177,8 @@ def _perform_login(self, username, password):
 
     def _real_extract(self, url):
         lang, video_id = self._match_valid_url(url).group('lang', 'id')
+        if not lang:
+            lang = 'fr'
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:181]:
```suggestion
```

File: yt_dlp/extractor/adn.py [hunk 5]
```diff
@@ -311,6 +313,8 @@ class ADNSeasonIE(ADNBaseIE):
 
     def _real_extract(self, url):
         lang, video_show_slug = self._match_valid_url(url).group('lang', 'id')
+        if not lang:
+            lang = 'fr'
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:317]:
```suggestion
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/adn.py [hunk 6]
```diff
@@ -327,8 +328,9 @@ def _real_extract(self, url):
 
         def entries():
             for episode_id in traverse_obj(episodes, ('videos', ..., 'id', {str_or_none})):
-                yield self.url_result(
-                    f'https://animationdigitalnetwork.{lang}/video/{video_show_slug}/{episode_id}',
-                    ADNIE, episode_id)
+                baseurl = 'https://animationdigitalnetwork.com' + ('/' + lang if lang != 'fr' else '')
```
Comment (bashonly) [inline yt_dlp/extractor/adn.py:331]:
```suggestion
```

**Review by bashonly — APPROVED** [review]
I applied some changes, please test them and let me know what you think

**Comment by infanf** [comment]
Treated them, works great. Thanks

**Pull request merged**